### ISSUE TITLE
[C#] Implement new Concept Exercise: interfaces

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#1225]
This issue describes how to implement the `interfaces` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Interfaces in C#.

## Learning objectives

- Know what interfaces are.
- Know how to use interfaces.
- Know how to define an interface.
- Know how to implement an interface.
- Know how to explicitly implement an interface.

## Out of scope

- Default interface methods.

## Concepts

The Concepts this exercise unlocks are:

- `interfaces`: know what interfaces are; know how to use interfaces; know how to define an interface; know how to implement an interface; know how to explicitly implement an interface.

## Prequisites

This exercise's prerequisites Concepts are:

- `classes`: know how to work with classes.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Interface][interface]: what an interface is and how to implement it.
- [Explicit interface implementation][explicit-interface-implementation]: when and how to use explicit interface implementations.

### After

- [Interface][interface]: what an interface is and how to implement it.
- [Explicit interface implementation][explicit-interface-implementation]: when and how to use explicit interface implementations.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[implementing-a-concept-exercise]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[interface]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/
[explicit-interface-implementation]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation

**Comment by senal** [comment]
Hi @ErikSchierboom,
I've assinged this task fort me but noticed the implementation of his exercise has been commited to master. :(
Should we require to close this task?
Cheers

**Comment by ErikSchierboom** [comment]
@senal It's great that you're working on this! I've removed the `status/help-wanted` label and added the `status/in-progress` label. Custom has it that the person that wants to work on a issue also posts a comment in this issue.

Regarding the PR. We're currently working on a strategy to make PR's simpler, to make the PR process more streamlined and fun for both reviewer and reviewee. The main idea is that instead of building out the whole exercise beforehand, the PR submitter submits an intial (draft) PR with only a couple of files. We're working on determining which files that will be, but most likely it will be an example implementation file and a `meta/design.md` or `introduction.md` file. Could you try to open a rather minimal, draft PR according to what I just mentioned when you're ready too? In other words: don't try to build the entire exercise at once, but let's do that in multiple steps :)

We'll have proper tooling for this later.

**Comment by senal** [comment]
@ErikSchierboom ,
I'm ready with my draft PR, which remote branch do I need to refere in my PR? Master or any other branch.
please advise.

**Comment by ErikSchierboom** [comment]
@senal PR's should be merge into the `master` branch.

**Comment by mikedamay** [comment]
See PR #1472 for more history on this issue.

**Comment by mikedamay** [comment]
closed by PR  #2076
